### PR TITLE
Cleans up API Token Auth Code

### PR DESCRIPTION
**Pull request by adambray**
- ensures a blank (empty string) token won't be checked against DB
- ensures all new users get an API token
- provides a rake task to auto generate api tokens for all existing users

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def set_current_user model
 
     def current_user
       begin
-        if params[:api_token]
+        if params[:api_token].present?
```
Comment (jshawl) [inline app/controllers/application_controller.rb]: Can you write a test showing this bug in action

File: lib/tasks/api_tokens.rake [hunk 2]
```diff
@@ -0,0 +1,7 @@
+namespace :api do
+  desc "Generates API tokens for all users which don't have one"
```
Comment (jshawl) [inline lib/tasks/api_tokens.rake]: Why rake task instead of migration?

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (adambray) [inline app/controllers/application_controller.rb]: @jshawl done!

File: lib/tasks/api_tokens.rake (hunk 2 above)
Comment (adambray) [inline lib/tasks/api_tokens.rake]: @jshawl I'm honestly on the fence about this... I'm starting to feel like small things like this are better done in a rake to keep migrations clean.

Comment (adambray) [inline lib/tasks/api_tokens.rake]: Also, it'll encourage us to come up with a good way to run rake tasks on wdidc.org

File: app/models/user.rb [hunk 3]
```diff
@@ -123,10 +125,6 @@ def as_json(options={})
     super.as_json(except: :password_digest)
   end
 
-  def generate_api_token
-    update(api_token: Digest::MD5.hexdigest(self.name + Time.now.to_s))
```
Comment (adambray) [inline app/models/user.rb]: @jshawl I believe this was insecure btw... knowing a user's name, you could just iterate over all reasonable values of the time their token was generated, and keep trying until you find it.

Comment (jshawl) [inline app/models/user.rb]: oh wow good point.. oops

File: lib/tasks/api_tokens.rake (hunk 2 above)
Comment (jshawl) [inline lib/tasks/api_tokens.rake]: `garnet run ...` ?

**Comment by jshawl** [comment]
:+1:

**Comment by nolds9** [comment]
:+1:

**Pull request merged**